PR: https://github.com/odoo/odoo/pull/

From: 0ebc5b7233d517f916f5e4196be51ed669b72633
From: Yannick Tivisse
Date: 2019-11-05 12:08:03

Structural Changes: 1
Total Changes: 34

[IMP] base_automation: Adapt tests to work with/without demo data

================================= pseudo patch: =================================

--- a/addons/base_automation/models/__init__.py
+++ b/addons/base_automation/models/__init__.py
@@ -3,4 +3,3 @@
 
 from . import base_automation
 from . import ir_actions
-from . import ir_demo

--- a/addons/base_automation/models/base_automation.py
+++ b/addons/base_automation/models/base_automation.py
@@ -94,13 +94,14 @@ class BaseAutomation(models.Model):
                 }
             }}
 
-    @api.model
-    def create(self, vals):
-        vals['usage'] = 'base_automation'
-        base_automation = super(BaseAutomation, self).create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            vals['usage'] = 'base_automation'
+        base_automations = super(BaseAutomation, self).create(vals_list)
         self._update_cron()
         self._update_registry()
-        return base_automation
+        return base_automations
 
     def write(self, vals):
         res = super(BaseAutomation, self).write(vals)

--- a/addons/base_automation/models/ir_demo.py
+++ b/None
@@ -1,22 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models
-
-
-class IrDemo(models.TransientModel):
-    _inherit = 'ir.demo'
-
-    def install_demo(self):
-        # Prevent the registry to reload while loading demo data, since `create` calls
-        # `_update_registry`.
-        self.pool.ready = False
-        try:
-            result = super().install_demo()
-        finally:
-            self.pool.ready = True
-
-        # Reload the registry
-        self.env['base.automation']._update_registry()
-
-        return result
